src/commands/switch.js

```diff
@@ -19,6 +24,21 @@ class SwitchCommand extends Command {
       }
     })
 
+    if(Object.keys(flags).length) {
+
+      selectedAccount = Object.entries(availableUsersChoices).find(([k, v]) => Object.keys(flags)[0] === 'userId' ?
+          k === flags[Object.keys(flags)[0]] : v.includes(flags[Object.keys(flags)[0]]))
```

Comment:
This logic should be broken down so the imperative check for 'email' is explicit.